kayenta-influxdb/src/main/java/com/netflix/kayenta/influxdb/config/InfluxdbResponseConverter.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2018 Joseph Motha
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.kayenta.influxdb.config;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.kayenta.influxdb.model.InfluxdbResult;
+
+import lombok.extern.slf4j.Slf4j;
+import retrofit.converter.ConversionException;
+import retrofit.converter.Converter;
+import retrofit.mime.TypedInput;
+import retrofit.mime.TypedOutput;
+
+@Component
+@Slf4j
+public class InfluxdbResponseConverter implements Converter {
+
+  private static final int DEFAULT_STEP_SIZE = 0;
+  private final ObjectMapper kayentaObjectMapper;
+
+  @Autowired
+  public InfluxdbResponseConverter(ObjectMapper kayentaObjectMapper) {
+    this.kayentaObjectMapper = kayentaObjectMapper;
+  }
+
+  @Override
+  public Object fromBody(TypedInput body, Type type) throws ConversionException {
+    
+    try (BufferedReader reader = new BufferedReader(new InputStreamReader(body.in()))) {
+      String json = reader.readLine();
+      log.debug("Converting response from influxdb: {}", json);
+      
+      Map result = getResultObject(json); 
+      List<Map> seriesList = (List<Map>) result.get("series");
+      
+      if (CollectionUtils.isEmpty(seriesList)) {
+        log.warn("Received no data from Influxdb.");
+        return null;
+      }
+      
+      Map series = seriesList.get(0);
+      List<String> seriesColumns = (List<String>) series.get("columns");
+      List<List> seriesValues = (List<List>) series.get("values");
+      List<InfluxdbResult> influxdbResultsList = new ArrayList<InfluxdbResult>(seriesValues.size());
```

Comment:
Should be able to use the diamond syntax here:
s/new ArrayList<InfluxdbResult>/new ArrayList<>